### PR TITLE
Add optional JSON format to the ACL CLI commands output

### PR DESCRIPTION
**Pull request by mullakhmetov**
solve #7141

- Add command level formatter that incapsulates command output printing logiс that depends on the command `-format` option.
- Move `Print*` functions from `acl_helpers.go` into corresponding `prettyFormatter`.
- Add `jsonFormatter`.

### CODE REVIEW

File: command/acl/role/list/role_list.go [hunk 1]
```diff
@@ -53,8 +61,17 @@ func (c *cmd) Run(args []string) int {
...
-	for _, role := range roles {
-		acl.PrintRoleListEntry(role, c.UI, c.showMeta)
+	formatter, err := role.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatRoleList(roles)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/role/list/role_list.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/bootstrap/bootstrap.go [hunk 2]
```diff
@@ -41,13 +49,25 @@ func (c *cmd) Run(args []string) int {
...
 
-	acl.PrintToken(token, c.UI, false)
+	formatter, err := token.NewFormatter(c.format, false)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatToken(t)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/bootstrap/bootstrap.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/policy/create/policy_create.go [hunk 3]
```diff
@@ -114,13 +123,25 @@ func (c *cmd) Run(args []string) int {
...
 
-	aclhelpers.PrintPolicy(policy, c.UI, c.showMeta)
+	formatter, err := policy.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatPolicy(p)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/policy/create/policy_create.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/policy/list/policy_list.go [hunk 4]
```diff
@@ -53,8 +61,17 @@ func (c *cmd) Run(args []string) int {
...
-	for _, policy := range policies {
-		acl.PrintPolicyListEntry(policy, c.UI, c.showMeta)
+	formatter, err := policy.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatPolicyList(policies)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/policy/list/policy_list.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/policy/read/policy_read.go [hunk 5]
```diff
@@ -68,12 +77,25 @@ func (c *cmd) Run(args []string) int {
...
-	acl.PrintPolicy(policy, c.UI, c.showMeta)
+
+	formatter, err := policy.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatPolicy(p)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/policy/read/policy_read.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/policy/update/policy_update.go [hunk 6]
```diff
@@ -144,14 +154,25 @@ func (c *cmd) Run(args []string) int {
...
-	c.UI.Info(fmt.Sprintf("Policy updated successfully"))
-	acl.PrintPolicy(policy, c.UI, c.showMeta)
+	formatter, err := policy.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatPolicy(p)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/policy/update/policy_update.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/role/create/role_create.go [hunk 7]
```diff
@@ -101,13 +109,25 @@ func (c *cmd) Run(args []string) int {
...
 
-	aclhelpers.PrintRole(role, c.UI, c.showMeta)
+	formatter, err := role.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatRole(r)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/role/create/role_create.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/role/read/role_read.go [hunk 8]
```diff
@@ -58,35 +67,47 @@ func (c *cmd) Run(args []string) int {
...
 
-	acl.PrintRole(role, c.UI, c.showMeta)
+	formatter, err := role.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatRole(r)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/role/read/role_read.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/role/update/role_update.go [hunk 9]
```diff
@@ -156,43 +165,54 @@ func (c *cmd) Run(args []string) int {
...
-	c.UI.Info(fmt.Sprintf("Role updated successfully"))
-	acl.PrintRole(role, c.UI, c.showMeta)
+	formatter, err := role.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatRole(r)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/role/update/role_update.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/token/clone/token_clone.go [hunk 10]
```diff
@@ -61,14 +70,25 @@ func (c *cmd) Run(args []string) int {
...
-	c.UI.Info("Token cloned successfully.")
-	acl.PrintToken(token, c.UI, false)
+	formatter, err := token.NewFormatter(c.format, false)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatToken(t)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/token/clone/token_clone.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/token/clone/token_clone.go [hunk 11]
```diff
@@ -61,14 +70,25 @@ func (c *cmd) Run(args []string) int {
...
 	}
 
-	token, _, err := client.ACL().TokenClone(tokenID, c.description, nil)
+	t, _, err := client.ACL().TokenClone(tokenID, c.description, nil)
 	if err != nil {
 		c.UI.Error(fmt.Sprintf("Error cloning token: %v", err))
 		return 1
 	}
 
-	c.UI.Info("Token cloned successfully.")
```
Comment (mkeeler) [inline command/acl/token/clone/token_clone.go]: Good call removing this as its unnecessary when we print out the new token.

File: command/acl/token/create/token_create.go [hunk 12]
```diff
@@ -131,13 +140,25 @@ func (c *cmd) Run(args []string) int {
...
 
-	acl.PrintToken(token, c.UI, c.showMeta)
+	formatter, err := token.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatToken(t)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/token/create/token_create.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/token/list/token_list.go [hunk 13]
```diff
@@ -52,14 +60,17 @@ func (c *cmd) Run(args []string) int {
...
-		}
-		acl.PrintTokenListEntry(token, c.UI, c.showMeta)
+	formatter, err := token.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatTokenList(tokens)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/token/list/token_list.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/token/read/token_read.go [hunk 14]
```diff
@@ -59,28 +68,40 @@ func (c *cmd) Run(args []string) int {
...
 
-	acl.PrintToken(token, c.UI, c.showMeta)
+	formatter, err := token.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	out, err := formatter.FormatToken(t)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/token/read/token_read.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

File: command/acl/token/update/token_update.go [hunk 15]
```diff
@@ -236,38 +245,49 @@ func (c *cmd) Run(args []string) int {
...
-	acl.PrintToken(token, c.UI, c.showMeta)
+	formatter, err := token.NewFormatter(c.format, c.showMeta)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	out, err := formatter.FormatToken(t)
+	if err != nil {
+		c.UI.Error(err.Error())
```
Comment (mkeeler) [inline command/acl/token/update/token_update.go]:
```suggestion
		c.UI.Error(err.Error())
		return 1
```

**Comment by mullakhmetov** [comment]
@mkeeler thanks for review and suggestions. Status code fixed, documentation updated.

**Pull request merged**